### PR TITLE
Bump HeatpumpIR - add protocols

### PR DESCRIPTION
**Pull request by nagyrobi**
Adding protocols
```
airway
bgh_aud
panasonic_altdke
vaillantvai8
r51m
```
possible from HeatpumpIR to 1.0.27

## Description:

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#6996

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/climate/climate_ir.rst [hunk 1]
```diff
@@ -76,11 +78,11 @@ controller unit.
 
     # Example configuration entry
     remote_transmitter:
-      pin: GPIOXX
+      pin: GPIO32
```
Comment (jesserockz) [inline components/climate/climate_ir.rst]: Examples are not to have real pin numbers.
```suggestion
      pin: GPIOXX
```

File: components/climate/climate_ir.rst [hunk 2]
```diff
@@ -76,11 +78,11 @@ controller unit.
...
       carrier_duty_percent: 50%
 
     climate:
-      - platform: coolix       # adjust to match your AC unit!
+      - platform: coolix
```
Comment (jesserockz) [inline components/climate/climate_ir.rst]: Lets remove this so it forces the users to change it correctly
```suggestion
      - platform: REPLACEME
```

File: components/climate/climate_ir.rst [hunk 3]
```diff
@@ -165,7 +125,7 @@ IR receiver.
     remote_receiver:
       id: rcvr
       pin:
-        number: GPIOXX
+        number: GPIO14
```
Comment (jesserockz) [inline components/climate/climate_ir.rst]:
```suggestion
        number: GPIOXX
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

File: components/climate/climate_ir.rst [hunk 4]
```diff
@@ -39,12 +37,12 @@ submit a feature request (see FAQ).
...
 | :ref:`GREE<gree_ir>`                  | ``gree``            |                      |
 +---------------------------------------+---------------------+----------------------+
-| Hitachi                               | ``hitachi_ac344``   | yes                  |
+| Hitachi                               | ``hitachi_ac344``,  | yes                  |
 |                                       | ``hitachi_ac424``   |                      |
 +---------------------------------------+---------------------+----------------------+
 | :ref:`LG<climate_ir_lg>`              | ``climate_ir_lg``   | yes                  |
 +---------------------------------------+---------------------+----------------------+
-| :ref:`Midea<midea_ir>`                | ``midea_ir``        | yes                  |
+| Midea                                 | ``midea_ir``        | yes                  |
```
Comment (jesserockz) [inline components/climate/climate_ir.rst]: No need to remove the link to the lower section...
```suggestion
| :ref:`Midea<midea_ir>`                | ``midea_ir``        | yes                  |
```

File: components/climate/climate_ir.rst [hunk 5]
```diff
@@ -396,64 +335,61 @@ Configuration variables:
...
-    # Example configuration entry
-    climate:
-      - platform: gree
-        name: "AC"
-        sensor: room_temperature
-        model: yan
+.. _daikin_arc:
 
-.. _zhlt01:
+.. note::
```
Comment (jesserockz) [inline components/climate/climate_ir.rst]:
```suggestion
``daikin_arc`` Climate
-------------------------
```

File: components/climate/climate_ir.rst [hunk 6]
```diff
@@ -396,64 +335,61 @@ Configuration variables:
...
-- JBS
-- Turbo Air
-- Nakatomy
-- Celestial Air
-- Ager
-- Blueway
-- Airlux
-
-No additional configuration is required for this model.
+.. note::
```
Comment (jesserockz) [inline components/climate/climate_ir.rst]: We should leave the heading there so they populate correctly in the sidebar. Really I think these sections should stay in the same order as the table at the top please.

```suggestion
``zhlt01`` Climate
---------------------
```

File: components/climate/climate_ir.rst [hunk 7]
```diff
@@ -396,64 +335,61 @@ Configuration variables:
...
+
+    The ``zhlt01`` climate and protocol, based on the ZH/LT-01 remote controller, is used with many locally branded airconditioners, like: Eurom, Chigo, Tristar, Tecnomaster, Elgin, Geant, Tekno, Topair, Proma, Sumikura, JBS, Turbo Air, Nakatomy, Celestial Air, Ager, Blueway, Airlux, etc.
```
Comment (jesserockz) [inline components/climate/climate_ir.rst]:
```suggestion
The ``zhlt01`` climate and protocol, based on the ZH/LT-01 remote controller, is used with many locally branded airconditioners, like: Eurom, Chigo, Tristar, Tecnomaster, Elgin, Geant, Tekno, Topair, Proma, Sumikura, JBS, Turbo Air, Nakatomy, Celestial Air, Ager, Blueway, Airlux, etc.
```

File: components/climate/climate_ir.rst [hunk 8]
```diff
@@ -396,64 +335,61 @@ Configuration variables:
...
-      - platform: gree
-        name: "AC"
-        sensor: room_temperature
-        model: yan
+.. _daikin_arc:
 
-.. _zhlt01:
+.. note::
 
+    The Daikin ARC remotes (``daikin_arc`` climate, ``daikin_arc417``, ``daikin_arc480`` protocols) are used by the japanese model of Daikin.
```
Comment (jesserockz) [inline components/climate/climate_ir.rst]:
```suggestion
The Daikin ARC remotes (``daikin_arc`` climate, ``daikin_arc417``, ``daikin_arc480`` protocols) are used by the japanese model of Daikin.
```

**Review by jesserockz — CHANGES_REQUESTED** [review]

**Comment by nagyrobi** [comment]
@jesserockz the concept was to declutter the page by removing the empty sections of the entries which don't have any specific configuration, and move the notes to a common block at the end of the file. No reason to show something in the sidebar which doesn't have any specific configuration option, just a generic note.
They are in alphabetical order.

**Comment by nagyrobi** [comment]
Here's a new approach, inspired from `esp32_camera` and `ethernet`, where specific options are not listed on the TOC.
The various platform-specific information is linked from the table already.

File: components/climate/climate_ir.rst [hunk 9]
```diff
@@ -62,11 +60,15 @@ submit a feature request (see FAQ).
...
 +---------------------------------------+---------------------+----------------------+
+| :ref:`Arduino-HeatpumpIR<heatpumpir>` | ``heatpumpir``      |                      |
+| library                               |                     |                      |
++---------------------------------------+---------------------+----------------------+
 
 This component requires that you have configured a :doc:`/components/remote_transmitter`.
 
 Due to the unidirectional nature of IR remote controllers, this component cannot determine the
 actual state of the device and will assume the state of the device is the latest state requested.
+The assumed state is being restored at boot (if you set ``esp8266_restore_from_flash: true``).
```
Comment (jesserockz) [inline components/climate/climate_ir.rst]: `esp8266_restore_from_flash` is deprecated and will be removed in a future version. The correct option is `restore_from_flash` under the `esp8266` component.

**Comment by jesserockz** [comment]
> No reason to show something in the sidebar which doesn't have any specific configuration option, just a generic note.

Fair enough. Happy for it to not be in the sidebar. And its good now, the previous versions without "headers" would scroll down and because the notes were all just bunched at the bottom, you wouldnt actually know where it was scrolling to.

**Comment by nagyrobi** [comment]
> `esp8266_restore_from_flash` is deprecated and will be removed in a future version. The correct option is `restore_from_flash` under the `esp8266` component.

Corrected. Also please check out https://github.com/esphome/esphome-docs/pull/4040, https://github.com/esphome/esphome-docs/pull/4041, https://github.com/esphome/esphome-docs/pull/4042, https://github.com/esphome/esphome-docs/pull/4043.

**Review by jesserockz — APPROVED** [review]

**Pull request merged**